code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/history/AndroidEventHistoryDatabase.java

```diff
@@ -70,6 +67,30 @@ class AndroidEventHistoryDatabase implements EventHistoryDatabase {
         }
     }
 
+    @SuppressWarnings("checkstyle:NestedIfDepth")
+    private void openOrMigrateEventHistoryDatabaseFile() throws Exception {
```

Comment:
nit: Change the function signature to return a file to make it more readable. Return null in case of exception.

Reply:
Wondering if it would be better to return the file in the database folder instead of null. Historical data would be lost but new events would get stored going forward

Reply:
Discussed offline and returning file instead of null